packages/storage/src/metadata/gundbMetadataStore.ts

```diff
@@ -93,10 +105,15 @@ export class GundbMetadataStore implements UserMetadataStore {
   static async fromIdentity(
     username: string,
     userpass: string,
-    gunOrServer?: GunChainReference<GunDataState> | string,
+    gunOrServer?: GunInit | string,
   ): Promise<GundbMetadataStore> {
     const store = new GundbMetadataStore(username, userpass, gunOrServer);
-    await store.authenticateUser();
+
+    store._user = store.gunInit().user();
+    await store.authenticateUser(store._user, username, userpass);
+    store._publicUser = store.gunInit().user();
+    await store.authenticateUser(store._publicUser, PublicStoreUsername, PublicStorePassword);
```

Comment:
Can someone from public edit this?

Reply:
Yes, glad you asked. Currently someone from public should be able to modify this. After fixing the current issue, I plan to discuss how to make this more less publicly accessible.

Reply:
Maybe we have a remote administration gundb account and have it grant all users read permission but only that account can update the public records.

It's just that this solution would mean we have to put it behind an HTTP API or somewhere we can hide the credentials.